lib/actions/build.coffee

```diff
@@ -109,17 +109,15 @@ module.exports =
 		Logger = require('../utils/logger')
 
 		logger = new Logger()
-
 		logger.logDebug('Parsing input...')
 
-		Promise.try ->
-			# `build` accepts `[source]` as a parameter, but compose expects it
-			# as an option. swap them here
-			options.source ?= params.source
-			delete params.source
-
-			validateComposeOptions(sdk, options)
+		# `build` accepts `[source]` as a parameter, but compose expects it
+		# as an option. swap them here
+		options.source ?= params.source
+		delete params.source
 
+		Promise.resolve(validateComposeOptions(sdk, options))
+		.then ->
```

Comment:
The key fix in this PR, here and in `deploy.coffee`, is to await (`.then`) the promise returned by `validateComposeOptions` (which is declared `async`). `validateComposeOptions` loads the registry secrets from a file and assigns the result to `options['registry-secrets']`. Previously, by chance, sometimes this operation would be completed "in time" before `options['registry-secrets']` was accessed -- especially likely if the `--application` option was provided, in which case a network request is then awaited. But if `--application` was not given, the chance would increase that the registry secrets would not be parsed in time, and authentication would fail as experienced by a customer.